tests/integration/pilot/common/routing.go

```diff
@@ -96,13 +96,17 @@ spec:
       request:
         add:
           istio-custom-header: user-defined-value`,
-				call: func() (echoclient.ParsedResponses, error) {
-					return podA.Call(echo.CallOptions{Target: apps.PodB[0], PortName: "http", Count: callCount})
-				},
-				validator: func(response echoclient.ParsedResponses) error {
-					return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
-						return ExpectString(response.RawResponse["Istio-Custom-Header"], "user-defined-value", "request header")
-					})
+				call: podA.CallWithRetryOrFail,
+				opts: echo.CallOptions{
+					Target:   apps.PodB[0],
+					PortName: "http",
+					Count:    callCount,
+					Validator: echo.ValidatorFunc(
+						func(response echoclient.ParsedResponses, _ error) error {
+							return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
```

Comment:
for future - maybe we can somehow optimize this so we don't have 
```
echo.ValidatorFunc(
func(response echoclient.ParsedResponses, _ error) error {
   return response.Check(func(_ int, response *echoclient.ParsedResponse) error {
```

boilerplate all over